providers/keap/parse.go

```diff
@@ -15,3 +17,70 @@ func makeNextRecordsURL(moduleID common.ModuleID) common.NextPageFunc {
 		return jsonquery.New(node).StrWithDefault("next_page_token", "")
 	}
 }
+
+// Before parsing the records if applicable API will be called one time to get model describing custom fields.
+// Object will then be enhanced using model.
+func (c *Connector) parseReadRecords(
+	ctx context.Context, config common.ReadParams, jsonPath string,
+) common.RecordsFunc {
+	return func(node *ajson.Node) ([]map[string]any, error) {
+		arr, err := jsonquery.New(node).Array(jsonPath, true)
+		if err != nil {
+			return nil, err
+		}
+
+		customFields, err := c.requestCustomFields(ctx, config.ObjectName)
```

Comment:
What happens when there are no custom fields on an object that we expect to have?

Reply:
Since, this is not so apparent I will change the code to be more explicit.

https://github.com/amp-labs/connectors/blob/4bdd68dcf7450bdb6c4716d4ba9888a10c621e53/providers/keap/parse.go#L67